src/Core/Resolvers/Sql Query Structures/SqlQueryStructure.cs

```diff
@@ -302,7 +302,7 @@ private List<OrderByColumn> PrimaryKeyAsOrderByColumns()
                 PaginationMetadata.Subqueries.Add(QueryBuilder.PAGINATION_FIELD_NAME, PaginationMetadata.MakeEmptyPaginationMetadata());
             }
 
-            EntityName = _underlyingFieldType.Name;
+            EntityName = GraphQLUtils.GetEntityNameFromContext(ctx);
```

Comment:
Is there any way to only have this change apply if querying DW database?

Reply:
Fixed.

Reply:
Actually, I dont see the need to special case it for only DW.. This is a query field right? So, shouldn't it be same irrespective of db type? Since, I thought the type change we are doing is for Mutation only.